src/iris/vendors/iris_slack.py

```diff
@@ -32,6 +32,26 @@ def __init__(self, config):
         self.sleep_range = config.get('sleep_range', 4)
         self.message_attachments = self.config.get('message_attachments', {})
 
+    def lookup_by_email(self,email):
+        lookup_endpoint = self.config['base_url'] + "/users.lookupByEmail"
+        payload = {'token': self.config['auth_token'], 'email': email}
+        try:
+            response = requests.post(lookup_endpoint,
+                                     data=payload,
+                                     proxies=self.proxy,
+                                     timeout=self.timeout)
+            if response.status_code == 200:
+                data = response.json()
+                if data.get('ok') and data.get('user'):
+                    if data['user'].get('id'):
+                        return data['user']['id']
+
+            logger.error('Failed resolve user id from email: %d', response.status_code)
+            return False
```

Comment:
I think it's better to surface the upstream error message to our users than returning false

Reply:
The user will get a 404 with a 'Failed to get id from Slack' message, not sure if additional specificity will be more useful to the end-user